docs-src/references/cluster-metrics.md

```diff
@@ -0,0 +1,174 @@
+Example: `sum(rate(poll_success{}[5m]))`
+
+### `poll_timeouts`
+
+Shows when no tasks are available for the poller within the poll timeout.
+Example: `sum(rate(poll_timeouts{}[5m]))`
+
+### `asyncmatch_latency`
+
+Measures async matched tasks from the time they are created to delivered. The larger this latency the longer tasks are sitting in the queue waiting for your workers to pick them up.
+Example: `histogram_quantile(0.95, sum(rate(asyncmatch_latency_bucket{service_name=~"matching"}[5m])) by (operation, le))`
+
+## History Service metrics
+
+History task is an internal task in Temporal that is created as part of a transaction to update Workflow state, and is processed by the Temporal History service.
+It is critical to ensure that the History task processing system is healthy.
+The following key metrics can be used to monitor the History Service health:
+
+### `task_requests`
+
+Emit on every task process request.
+Example: `sum(rate(task_requests{service="$service",operation=~"TransferActive.*"}[1m]))`
+
+### `task_errors`
+
+Emit on every task process error.
+Example: `sum(rate(task_errors{operation=~"TransferActive.*"}[1m]))`
+
+### `task_attempt`
+
+Number of attempts on each task execution. A task is retried forever, and each retry increases the attempt count.
+Example: `histogram_quantile($percentile, sum(rate(task_attempt_bucket{service="$service",operation=~"TransferActive.*"}[1m])) by (operation, le))`
+
+### `task_latency_processing`
+
+Shows the processing latency per attempt.
+Example: `histogram_quantile($percentile, sum(rate(task_latency_processing_bucket{operation=~"TransferActive.*",service="$service", service_name="history"}[1m])) by (operation, le))`
+
+### `task_latency`
+
+Measures the task processing latency for one attempt.
+
+### `task_latency_queue`
+
+Measures the duration, end-to-end, from when the task should be executed (from the time it was fired), to when the task is done.
+
+### `task_latency_load`
+
+(Available only in v1.18.0+) Measures the duration from task generation to task loading (task schedule to start latency for persistence queue).
+
+### `task_latency_schedule`
+
+(Available only in v1.18.0+) Measures the duration from task submission (to task scheduler) to processing (task schedule to start latency for in-memory queue).
+
+### `queue_latency_schedule`
+
+(Available only in v1.18.0+) Measures the time it takes to schedule 100 tasks in one task channel in the host-level task scheduler.
+If there are less than 100 tasks in the task channel for 30 sesoncds, the latency will be scaled to 100 tasks upon emission.
+Note: This is still an experimental metric and is subject to change.
+
+### `task_latency_userlatency`
+
+Shows the latency introduced because of Workflow logic. For example, if you have one Workflow scheduling many Activities or Child Workflows at the same time, it can cause a per-Workflow lock contention. The wait period for the per-Workflow lock is counted as `userlatency`.
```

Comment:
```suggestion
Shows the latency introduced because of Workflow logic.
For example, if you have one Workflow scheduling many Activities or Child Workflows at the same time, it can cause a per-Workflow lock contention.
The wait period for the per-Workflow lock is counted as `userlatency`.
```